### ISSUE TITLE
opentelemetry-jaeger and WASM

### ISSUE THREAD

**Issue by jesseditson** [open-telemetry/opentelemetry-rust#1235]
Apologies if I'm missing something simple, but I'm trying to set up a client-side exporter that will send telemetry data to a jaeger instance.

I found the `wasm_collector_client` flag on `opentelemetry-jaeger`, and it appeared to be what I needed - however I can't find any examples or tests with it configured and running. Based on existing docs, it looks like the way to set it up would be something along the lines of:

```
global::set_text_map_propagator(opentelemetry_jaeger::Propagator::new());
let tracer = opentelemetry_jaeger::new_wasm_collector_pipeline()
        .with_endpoint("http://localhost:16686/api/traces")
        .install_batch(opentelemetry::runtime::AsyncStd)
        .expect("Failed setting up tracer");
```

However, to enable this I'd have to turn on the `rt-async-std` flag - this allows `cargo check` to pass, but if you target `wasm32-unknown-unknown` the build fails in a bunch of ways - the first few being:

```
error[E0425]: cannot find function `spawn` in module `async_std::task`
   --> .cargo/registry/src/index.crates.io-6f17d22bba15001f/opentelemetry_sdk-0.20.0/src/runtime.rs:128:34
    |
128 |         let _ = async_std::task::spawn(future);
    |                                  ^^^^^ not found in `async_std::task`
    |
help: consider importing this function
    |
9   + use std::thread::spawn;
    |
help: if you import `spawn`, refer to it directly
    |
128 -         let _ = async_std::task::spawn(future);
128 +         let _ = spawn(future);
    |

error[E0277]: `*mut u8` cannot be sent between threads safely
   --> .cargo/registry/src/index.crates.io-6f17d22bba15001f/opentelemetry_sdk-0.20.0/src/runtime.rs:119:21
    |
119 |     type Interval = async_std::stream::Interval;
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `*mut u8` cannot be sent between threads safely
    |
    = help: within `Interval`, the trait `std::marker::Send` is not implemented for `*mut u8`
```

It seems like perhaps the runtime crate isn't compatible with `wasm32` targets - but if this is the case, how is `wasm_collector_client` supposed to work at all? It appears that a runtime is required for it to function, and `tokio` doesn't support `wasm32`. Perhaps I should be providing a custom runtime? Are there any examples of `wasm_collector_client` working somewhere?

**Comment by rymbdrr** [comment]
Thanks for raising this. I checked the current `main` (0.31.x), and the situation has changed since this issue was opened:

- The old `opentelemetry-jaeger` exporter pipeline API (including `new_wasm_collector_pipeline` / `wasm_collector_client`) is no longer present in this repo.
- The Jaeger crate in this workspace is now `opentelemetry-jaeger-propagator` (propagation only).
- Exporting to Jaeger is done via OTLP (`opentelemetry-otlp`) rather than a dedicated Jaeger exporter path.

You can also see this direction in the current `opentelemetry-jaeger-propagator` README: it explicitly says to use `opentelemetry-otlp` for exporting telemetry to Jaeger.

Given that, I think this specific issue (about `new_wasm_collector_pipeline` + `rt-async-std` on `wasm32`) is outdated as written and likely not actionable in its original form.

If there is still a need for **browser wasm32 -> Jaeger via OTLP** guidance/support, we should track that as an OTLP-focused issue (docs + tested example), since that is the current exporter path.

**Comment by jesseditson** [comment]
Yes I no longer am using Jaeger in the setup since updating. Closing!
